src/test/java/core/basesyntax/service/TransactionHandlerImplTest.java

```diff
@@ -0,0 +1,91 @@
+            OperationHandler> handlersMap = new HashMap<>();
+
+    @Before
+    public void setUp() throws Exception {
+        fruitTransactions = new ArrayList<>();
+        expectedFruits = new HashMap<>();
+        handlersMap.put(FruitTransaction.Operation.BALANCE, new BalanceHandlerImpl());
+        handlersMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandlerImpl());
+        handlersMap.put(FruitTransaction.Operation.RETURN, new ReturnHandlerImpl());
+        handlersMap.put(FruitTransaction.Operation.SUPPLY, new SupplyHandlerImpl());
+        TransactionStrategy transactionStrategy = new TransactionStrategyImpl(handlersMap);
+        transactionHandler = new TransactionHandlerImpl(transactionStrategy);
+    }
+
+    @Test
+    public void handle_Ok() {
+        FruitTransaction fruitTransaction1 = new FruitTransaction("banana",
+                FruitTransaction.Operation.getByCode("b"), 20);
+        fruitTransactions.add(fruitTransaction1);
+        FruitTransaction fruitTransaction2 = new FruitTransaction("apple",
+                FruitTransaction.Operation.getByCode("b"), 100);
+        fruitTransactions.add(fruitTransaction2);
+        FruitTransaction fruitTransaction3 = new FruitTransaction("banana",
+                FruitTransaction.Operation.getByCode("s"), 100);
+        fruitTransactions.add(fruitTransaction3);
+        FruitTransaction fruitTransaction4 = new FruitTransaction("banana",
+                FruitTransaction.Operation.getByCode("p"), 13);
+        fruitTransactions.add(fruitTransaction4);
+        FruitTransaction fruitTransaction5 = new FruitTransaction("apple",
+                FruitTransaction.Operation.getByCode("r"), 10);
+        fruitTransactions.add(fruitTransaction5);
+        FruitTransaction fruitTransaction6 = new FruitTransaction("apple",
+                FruitTransaction.Operation.getByCode("p"), 20);
+        fruitTransactions.add(fruitTransaction6);
+        FruitTransaction fruitTransaction7 = new FruitTransaction("banana",
+                FruitTransaction.Operation.getByCode("p"), 5);
+        fruitTransactions.add(fruitTransaction7);
+        FruitTransaction fruitTransaction8 = new FruitTransaction("banana",
+                FruitTransaction.Operation.getByCode("s"), 50);
+        fruitTransactions.add(fruitTransaction8);
+        transactionHandler.handle(fruitTransactions);
+        expectedFruits.put("banana", 152);
+        expectedFruits.put("apple", 90);
+        assertEquals(expectedFruits.size(), Storage.fruits.size());
+        assertEquals(expectedFruits, Storage.fruits);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void handle_nullInput_notOk() {
+        transactionHandler.handle(null);
+    }
+
+    @Test
+    public void handle_emptyInputList_ok() {
+        transactionHandler.handle(fruitTransactions);
+        assertEquals(expectedFruits.size(), Storage.fruits.size());
+        assertEquals(expectedFruits, Storage.fruits);
+    }
```

Comment:
check that storage is really empty - size == 0. and your list is not empty because of static test data